pkg/mco/mcplist.go

```diff
@@ -0,0 +1,138 @@
+package mco
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func ListMCP(apiClient *clients.Settings, listOptions metav1.ListOptions) ([]*MCPBuilder, error) {
+	glog.V(100).Infof("Listing all MCP resources with the options %v", listOptions)
+
+	mcpList, err := apiClient.MachineConfigPools().List(context.Background(), listOptions)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list MCP objects due to %s", err.Error())
+
+		return nil, err
+	}
+
+	var mcpObjects []*MCPBuilder
+
+	for _, mcp := range mcpList.Items {
+		copiedMcp := mcp
+		mcpBuilder := &MCPBuilder{
+			apiClient:  apiClient,
+			Object:     &copiedMcp,
+			Definition: &copiedMcp,
+		}
+
+		mcpObjects = append(mcpObjects, mcpBuilder)
+	}
+
+	return mcpObjects, nil
+}
+
+func ListMCPByMachineConfigSelector(apiClient *clients.Settings, mcpLabel string) (*MCPBuilder, error) {
+	glog.V(100).Infof("GetByLabel returns MachineConfigPool with the specified label: %v", mcpLabel)
+
+	mcpList, err := ListMCP(apiClient, metav1.ListOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mcp := range mcpList {
+		for _, label := range mcp.Object.Spec.MachineConfigSelector.MatchExpressions {
+			for _, value := range label.Values {
+				if value == mcpLabel {
+					return mcp, nil
+				}
+			}
+		}
+
+		for _, label := range mcp.Object.Spec.MachineConfigSelector.MatchLabels {
+			if label == mcpLabel {
+				return mcp, nil
+			}
+		}
+	}
+
+	return nil, fmt.Errorf("cannot find MachineConfigPool that targets machineConfig with label: %s", mcpLabel)
+}
+
+func ListMCPWaitToBeStableFor(apiClient *clients.Settings, stableDuration, timeout time.Duration) error {
```

Comment:
I think nesting two wait.PollImmediate functions and using two timeout arguments is confusing and can lead to unexpected behavior. I would prefer to just have one timeout and one wait.PollImmediate function @kononovn  WDYT?

Reply:
It's impossible for list. Please check function logic carefully. We do need nested timeout because we iterate over list and if even one MCP is not stable we reset second timer.